datumaro/plugins/ndr.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+import cv2
+import numpy as np
+from scipy.linalg import orth
+
+from datumaro.components.extractor import Transform, DEFAULT_SUBSET_NAME
+
+class NDR(Transform):
+    """
+    Near-duplicated image removal |n
+    Removes near-duplicated images in subset |n
+    """
+    def __init__(self, extractor, working_subset=None, duplicated_subset="duplicated", algorithm='gradient',
+            num_cut=None, over_sample='random', under_sample='uniform', seed=None, **kwargs):
+        """
+        Near-duplicated image removal
+
+        Arguments
+        ---------------
+        working_subset: str
```

Comment:
I'm not sure if using a parameter is better than passing just a subset as an input. Also, I think, it looks more natural, when `None` run the algorithm for the whole dataset.

Reply:
I think that might return unexpected results if user has a subset like *val* or *test*.